src/mixer.c

```diff
@@ -760,6 +760,51 @@ struct mixer_ctl *mixer_get_ctl_by_name_and_index(struct mixer *mixer,
     return NULL;
 }
 
+/** Gets an instance of mixer control handle, by the mixer control's name and device.
+ *  For instance, if two controls have same name, e.g. 'Playback Channel map', then PCM device returns the specific control.
+ * @param mixer An initialized mixer handle.
+ * @param name The control's name in the given mixer.
+ * @param device The PCM device
+ * @returns A handle to the mixer control.
+ * @ingroup libtinyalsa-mixer
+ */
+struct mixer_ctl *mixer_get_ctl_by_name_and_device(struct mixer *mixer,
+                                                  const char *name,
+                                                  unsigned int device)
+{
+    struct mixer_ctl_group *grp;
+    unsigned int n;
+    struct mixer_ctl *ctl;
+
+    if (!mixer || !name) {
+        return NULL;
+    }
+
+    if (mixer->h_grp) {
+        grp = mixer->h_grp;
+        ctl = grp->ctl;
+
+        for (n = 0; n < grp->count; n++) {
+            if ((!strcmp(name, (char*) ctl[n].info.id.name)) && (device == ctl[n].info.id.device)) {
+                return ctl + n;
+            }
+        }
+    }
+
+#ifdef TINYALSA_USES_PLUGINS
+    if (mixer->v_grp) {
+        grp = mixer->v_grp;
+        ctl = grp->ctl;
+
+        for (n = 0; n < grp->count; n++)
+            if ((!strcmp(name, (char*) ctl[n].info.id.name)) && (device == ctl[n].info.id.device)) {
```

Comment:
No need to add one more pair of parentheses. This is also for line 788.

```
if (!strcmp(name, (char*) ctl[n].info.id.name) &&
        device == ctl[n].info.id.device) {
```

Reply:
fixed at both places